### ISSUE TITLE
Modifying task name under "EXPERIMENTS" in the UI does not reflect int the model checkpoints name under "MODELS"

### ISSUE THREAD

**Issue by nizamreplica** [allegroai/clearml#1179]
## Describe the bug
If one were to change the task name under experiments during a run, the change is not reflected in the model name prefix when checkpoints are generated under models
![Screenshot 2023-12-22 at 2 02 07 pm](https://github.com/allegroai/clearml/assets/140988784/85e5d01b-09fb-4329-8348-2fecd6a57a24)

This would make sense if the task has aborted or completed, but should work when it is running.

**Note** that the image attachments for the sake of this issue description are on `status=ABORTED` but the bug was found on `status=RUNNING`

## To reproduce
1) Start a task that saves something as model checkpoints under models
2) While it's still running edit the task name in ClearML UI
![Screenshot 2023-12-22 at 7 18 32 pm](https://github.com/allegroai/clearml/assets/140988784/f5484d69-cab3-4e46-a3c4-02d50bca5be5)
3) Now instead of model checkpoints being saved with prefix `test_base_model_3`, it still saves them as `test_base_model_2`

## Expected behaviour
The model checkpoints should be saved with prefix `test_base_model_3`

## Environment
Server type (self hosted)
ClearML SDK Version: 1.13.2
ClearML Server Version: 1.12.1-397
Python Version: 3.10.13
OS (Linux, Ubuntu)

**Comment by ainoam** [comment]
@nizamreplica Indeed, ClearML does not provide UI control of tasks while they are executed. You can only control a tasks configuration pre-execution ('draft') and modify descriptive elements (e.g. name, tags, description etc.) post execution.

Manually changing the tasks behaviour as it executes could lead to incoherent results, and prevent reproducibility and so is not supported. Does this make sense?
